src/main/java/hudson/plugins/git/GitSCM.java

```diff
@@ -1767,22 +1766,17 @@ public BuildData copyBuildData(Run build) {
      * @return the last recorded build data
      */
     public @CheckForNull BuildData getBuildData(Run build) {
-        BuildData buildData = null;
         while (build != null) {
             List<BuildData> buildDataList = build.getActions(BuildData.class);
             for (BuildData bd : buildDataList) {
                 if (bd != null && isRelevantBuildData(bd)) {
-                    buildData = bd;
-                    break;
+                    return bd;
```

Comment:
This changes the semantics of the `getBuildData` call.  Previously it return the last recorded build data, now it returns the first.

Is that intentional?  How does that help with the problem?